### PR TITLE
Backport 20996 remove password reset

### PR DESCRIPTION
**Pull request by nickvergessen**
Backport of #20996 and #21148

Approval https://github.com/owncloud/core/pull/20996#issuecomment-165476689

@MorrisJobke @icewind1991 @blizzz

### CODE REVIEW

**Comment by PVince81** [comment]
:+1:

**Comment by bboule** [comment]
@gig13 @MorrisJobke Was there a user waiting for this?  I cant remember which one?

**Comment by MorrisJobke** [comment]
> @gig13 @MorrisJobke Was there a user waiting for this? I cant remember which one?

00002757 - https://github.com/owncloud/core/issues/16816

**Comment by nickvergessen** [comment]
@MorrisJobke second +1?

File: lib/private/util.php [hunk 1]
```diff
@@ -947,6 +947,16 @@ public static function displayLoginPage($errors = array(), $messages = []) {
 			$parameters['redirect_url'] = $_REQUEST['redirect_url'];
 		}
 
+		$parameters['canResetPassword'] = true;
+		if (!\OC::$server->getSystemConfig()->getValue('lost_password_link')) {
+			if (isset($_REQUEST['user'])) {
+				$user = \OC::$server->getUserManager()->get($_REQUEST['user']);
+				if ($user instanceof IUser) {
```
Comment (blizzz) [inline lib/private/util.php]: IUser is undefined, the use statement is missing.

Comment (nickvergessen) [inline lib/private/util.php]: Fixed

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**